corehq/apps/domain/tests/test_forms.py

```diff
@@ -29,119 +21,239 @@ def test_visible_fields(self):
+        form = self._create_form()
+        visible_field_names = self._get_visible_fields(form)
         self.assertIn('secure_sessions_timeout', visible_field_names)
 
     def test_advanced_domain_security(self):
         self.mock_domain_has_privilege.return_value = True
-        form = self.create_form()
-        visible_field_names = self.get_visible_fields(form)
+        form = self._create_form()
+        visible_field_names = self._get_visible_fields(form)
         advanced_security_fields = {'ga_opt_out', 'strong_mobile_passwords', 'two_factor_auth', 'secure_sessions'}
         self.assertTrue(advanced_security_fields.issubset(set(visible_field_names)))
 
-    def create_form(self):
+# Helpers
+    def setUp(self):
+        super().setUp()
+        patcher = patch.object(forms, 'domain_has_privilege')
+        self.mock_domain_has_privilege = patcher.start()
+        self.mock_domain_has_privilege.return_value = False
+        self.addCleanup(patcher.stop)
+
+    def _create_form(self):
         return PrivacySecurityForm(user_name='test_user', domain='test_domain')
 
-    def get_visible_fields(self, form):
+    def _get_visible_fields(self, form):
         fieldset = form.helper.layout.fields[0]
         return [field[0] for field in fieldset.fields]
 
 
-class TestDomainGlobalSettingsForm(TestCase):
+class DomainGlobalSettingsFormTests(SimpleTestCase):
+    def test_all_visible_fields(self):
+        form = self._create_form()
+        self.assertEqual([
+            'hr_name',
+            'project_description',
+            'default_timezone',
+            'default_geocoder_location',
+            'logo',
+            'delete_logo',
+            'call_center_enabled',
+            'call_center_type',
+            'call_center_case_owner',
+            'call_center_case_type',
+            'mobile_ucr_sync_interval',
+            'confirmation_link_expiry'
+        ], form.get_visible_field_names())
 
-    def setUp(self) -> None:
-        super().setUp()
-        self.domain = Domain(name='test_domain')
-        self.call_settings = OperatorCallLimitSettings(domain='test_domain')
-        self.call_settings.save()
-        self.domain.save()
-
-    def test_confirmation_link_expiry_not_present_when_flag_not_set(self):
-        set_toggle(TWO_STAGE_USER_PROVISIONING_BY_SMS.slug, self.domain, False, namespace=NAMESPACE_DOMAIN)
-        form = self.create_form()
-        self.assertTrue('confirmation_link_expiry' not in form.fields)
-
-    def test_confirmation_link_expiry_default_present_when_flag_set(self):
-        set_toggle(TWO_STAGE_USER_PROVISIONING_BY_SMS.slug, self.domain, True, namespace=NAMESPACE_DOMAIN)
-        form = self.create_form(confirmation_link_expiry=self.domain.confirmation_link_expiry_time)
-        form.full_clean()
-        form.save(Mock(), self.domain)
-        self.assertTrue('confirmation_link_expiry' in form.fields)
-        self.assertEqual(168, self.domain.confirmation_link_expiry_time)
```

Comment:
It's hard for me to determine if these tests were refactored or removed, possibly because they were also renamed at the same time. Are there new tests that cover what these were doing?

Reply:
Yes. The first test is covered by `test_if_sms_user_provisioning_disabled_field_is_removed`. The second is generally covered by `test_all_visible_fields`, as that tests the base case where all fields should be included. `test_confirmation_link_expiry_default_present_when_flag_set` was testing too many things, and so the save behavior was split out into `test_save_writes_confirmation_link_expiry`.